ocsp.go

```diff
@@ -598,6 +577,9 @@ func getAllRevocationStatus(verifiedChains []*x509.Certificate) []*ocspStatus {
 	results := make([]*ocspStatus, n)
 	for j := 0; j < n; j++ {
 		results[j] = getRevocationStatus(verifiedChains[j], verifiedChains[j+1])
+		if !isValidOCSPStatus(results[j].code) {
```

Comment:
This would skip further validation only if the Certificate Status of the OCSP Response is invalid. Ideally the only case where we should continue checking OCSP Responses for other Certificates is if the current OCSP Response has certificate status == Good. For ex, if even a single certificate in the chain is Revoked the whole chain is untrustworthy and should be discarded.

Reply:
Other than this comment .. LGTM.

Reply:
Actually, there is one more thing - the fail open log? I don't see the Fail Open specific Logging. Something like this - https://devtools1.int.snowflakecomputing.com/fisheye/browse/svn/Snowflake/trunk/Python/snowflake/connector/ocsp_snowflake.py?hb=true#to963

Reply:
ah, ok, let me add the log.

Reply:
@h-chaturvedi , the last commit should include all changes.
Btw, can you write up the functional spec including how our drivers uses OCSP?

Reply:
Sure. Created task - SNOW-81523.